dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/WebHooks/WebHooksEndToEndTests.cs

```diff
@@ -0,0 +1,440 @@
+            EmailAddress = Faker.Internet.Email(),
+            FirstName = Faker.Name.First(),
+            MiddleName = Faker.Name.Middle(),
+            LastName = Faker.Name.Last(),
+            PreferredName = Faker.Name.FullName(),
+            MobileNumber = Faker.Phone.Number(),
+            Trn = "7754311",
+            TrnLookupStatus = TrnLookupStatus.Found,
+        };
+
+        var mergedUserId = Guid.NewGuid();
+
+        // Act
+        var userCreatedNotification = new NotificationEnvelope()
+        {
+            NotificationId = Guid.NewGuid(),
+            Message = new UserMergedMessage()
+            {
+                MasterUser = user,
+                MergedUserId = mergedUserId
+            },
+            MessageType = UserMergedMessage.MessageTypeName,
+            TimeUtc = HostFixture.Clock.UtcNow
+        };
+
+        await notificationPublisher.PublishNotification(userCreatedNotification);
+
+        // Assert
+        if (expectedToReceiveMessage)
+        {
+            WebHookRequestObserver.AssertWebHookRequestsReceived(
+            r =>
+            {
+                Assert.NotNull(r.ContentType);
+                Assert.NotNull(r.Signature);
+                Assert.NotNull(r.Body);
+                Assert.Equal("application/json", r.ContentType);
+                var expectedSignature = GenerateExpectedSignature(secret, r.Body);
+                Assert.Equal(expectedSignature, r.Signature);
+
+                var expectedJson = JsonSerializer.SerializeToNode(new
+                {
+                    notificationId = userCreatedNotification.NotificationId,
+                    message = new
+                    {
+                        masterUser = new
+                        {
+                            userId = user.UserId,
+                            dateOfBirth = user.DateOfBirth,
+                            emailAddress = user.EmailAddress,
+                            firstName = user.FirstName,
+                            middleName = user.MiddleName,
+                            lastName = user.LastName,
+                            preferredName = user.PreferredName,
+                            mobileNumber = user.MobileNumber,
+                            trn = user.Trn,
+                            trnLookupStatus = user.TrnLookupStatus.ToString()
+                        },
+                        mergedUserId = mergedUserId
+                    },
+                    messageType = UserMergedMessage.MessageTypeName,
+                    timeUtc = HostFixture.Clock.UtcNow
+                });
+
+                AssertEx.JsonEquals(
+                    expectedJson,
+                    JsonNode.Parse(r.Body));
+            });
+        }
+        else
+        {
+            WebHookRequestObserver.AssertWebHookRequestsReceived(Array.Empty<Action<WebHookRequest>>());
+        }
+    }
+
+    private static string GenerateExpectedSignature(string secret, string content)
```

Comment:
Could make the `CalculateSignature` method on `WebHookNotificationSender` `public static` and use that instead of duplicating here.

Reply:
Done